testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1005,20 +1005,34 @@ impl BitcoinRegtestController {
             }
         }
 
-        // Stop as soon as the fee_rate is 1.25 higher, stop RBF
+        // Stop as soon as the fee_rate is 1.50 higher, stop RBF
         if ongoing_op.fees.fee_rate > (self.config.burnchain.satoshis_per_byte * 150 / 100) {
-            let res = self.send_block_commit_operation(payload, signer, None, None, None, &vec![]);
-            return res;
+            warn!("RBF'd block commits reached 1.5x satoshi per byte fee rate, not resubmitting");
+            self.ongoing_block_commit = Some(ongoing_op);
+            return None;
+        }
+
+        // Did a re-org occurred since we fetched our UTXOs, or are the UTXOs so stale that they should be abandoned?
+        let mut traversal_depth = 0;
+        let mut burn_chain_tip = burnchain_db.get_canonical_chain_tip().ok()?;
+        let mut found_last_mined_at = false;
+        while traversal_depth < 6 {
```

Comment:
This should probably be a constant somewhere, not `6`.

Reply:
Fixed in https://github.com/blockstack/stacks-blockchain/pull/2454/commits/bb3171d7cd2173c063a72c57dab7e69fdea23bea

Reply:
I trust your tests, but do you mind explaining what was wrong with the previous logic? I'm not sure to see how it was wrong.

Reply:
Yep, the prior logic had a few issues:

1. It should have been using the "stubbed" sortition ids -- those won't correspond to entries in the sortition db in any event. The stubbed sortition ids only exist for backwards compatibility while testing. In mainnet/testnet/regtest modes, the sortition id of a burn block will alway end up different than that stubbed id.
2. If the sortition is missing from the database, it wouldn't return an error, rather it would return `Ok(None)`.
3. The sortition db would continue to contain an entry for the "forked away" bitcoin blocks in most cases: data is deleted from it, and only PoX invalidations flip the "pox_valid" flag. if we wanted to use the sortition db to perform this check, we'd need to do something like `get_canonical_sortition_tip` to get the current sortition "chain tip", and then use that to invoke `get_sortition_id_for_bhh` which would look to see if it found the given burnchain tip within that sortition chain tip's fork history (it does this using a MARF lookup).

Reply:
Got it, thank you!